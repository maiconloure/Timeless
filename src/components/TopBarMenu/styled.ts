import styled from 'styled-components';

export const TopContainer = styled.div`
  grid-area: top;
  position: fixed;
  z-index: 9999;
  top: 0;
  width: 100vw;

  .tooltip {
    &:hover .tooltiptext {
      visibility: visible;
    }
  }

  .tooltiptext {
    font: 500 1.5rem Inter;
    visibility: hidden;
    min-width: 40px;
    max-width: 160px;
    background-color: black;
    color: #fff;
    text-align: center;
    border-radius: 3px;
    padding: 4px 8px;

    /* Position the tooltip */
    position: absolute;
    z-index: 99;
  }
`;

export const Bar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 70px;
  background-color: var(--color-primary-0);

  @media (min-width: 1000px) and (min-height: 768px) {
    height: 60px;
  }
`;

export const ProjectInfo = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Inter';

  h2,
  h3,
  h4 {
    display: none;
  }

  h3 {
    cursor: pointer;
  }

  img {
    width: 60px;
    border-radius: 6px;
    margin: 0 5px;
  }

  a {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  @media (min-width: 800px) {
    display: flex;
    align-items: center;

    h2 {
      margin-left: 5px;
      display: block;
      font-size: 1.8rem;
      font-weight: 800;
      text-transform: uppercase;
      color: #fff;
    }

    h3 {
      display: block;
      font-size: 1.6rem;
      color: #fff;
    }

    h4 {
      display: block;
      font-size: 2.4rem;
      font-weight: 900;
      color: #fff;
    }

    img {
      margin: 0 5px;
      width: 50px;
    }
  }
`;

export const User = styled.div`
  font-family: Inter;
  font-size: 1.4rem;
  margin: 0 10px 0 0;

  h2 {
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  p {
    text-align: right;
    font-weight: 600;
    color: #adff2f;
    margin: 0 4px 0 0;
  }

  @media (min-width: 1200px) and (min-height: 768px) {
    h2 {
      font-size: 2rem;
    }
    p {
      font-size: 1.7rem;
    }
  }
`;

export const ProfileIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  width: 55px;
  height: 55px;

  img {
    width: 60px;
    height: 60px;
    margin-right: 5px;
    border-radius: 50%;
  }

  @media (min-width: 1000px) and (min-height: 768px) {
    img {
      width: 55px;
      height: 55px;
    }
  }
`;

export const UserMenu = styled.div`
  position: absolute;
  top: 65px;
  left: 10px;
  min-width: 220px;
  height: 300px;
  padding: 10px;
  background: #ffffff;
  border-radius: 4px;
  color: #12254e;
  display: flex;
  flex-direction: column;
`;

export const UserInfo = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  color: #fff;
  width: 260px;
  max-width: 280;
`;

export const MenuOption = styled.div`
  border-top: 2px solid #dfe1e7;
  cursor: pointer;
  p {
    font: 500 1.5rem Roboto;
    opacity: 0.9;
    margin: 6px 0;
  }
`;

export const UserInfoMenu = styled.div`
  margin: 10px 5px;
  height: 70%;
  h2 {
    font: 500 2rem Inter;
  }

  h3 {
    font: 400 1.3rem Inter;
    padding: 2px 0;
    opacity: 0.7;
  }

  p {
    font: 600 1.4rem Inter;
    line-height: 1.6rem;
    padding: 4px 0;
    border-top: 2px solid #dfe1e7;
    opacity: 0.9;
  }
`;

export const MainUserMenu = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 2px;
  border-bottom: 2px solid #dfe1e7;

  h2 {
    font: 500 1.8rem Fira Code;
    opacity: 0.7;
  }

  img {
    cursor: pointer;
    width: 2rem;
  }
`;
