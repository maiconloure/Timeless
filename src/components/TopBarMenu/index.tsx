import React from 'react';
import { icons, images } from 'utils';

import { FixedMenuProps } from '../ComponentsInterface';
import * as St from './styled';

const FixedMenu = ({
  values: { user, currentBoard, toggleMenu },
  handlers: {
    handleLogout,
    handlerSaveBoard,
    handleOpenEditProfile,
    handleOpenEditBoard,
    handlerToggleBoard,
    handleToggleMenu,
  },
}: FixedMenuProps) => (
  <St.TopContainer>
    <St.Bar>
      <St.ProjectInfo>
        <a
          href="https://gitlab.com/capstoneproject-group-4/timeless"
          target="_blank"
          rel="noopener noreferrer">
          <div className="tooltip">
            <img src={images.logo} alt="Logo" />
            <span className="tooltiptext">Timeless salva automaticamente o seu trabalho!</span>
          </div>
          {/* <img src={icons.kenzie} alt="Project icon" /> */}
        </a>
        <div className="tooltip">
          <h2> Kenzie Academy Brasil </h2>
          <span className="tooltiptext">Nome da empresa</span>
        </div>
        <h4> &nbsp; | &nbsp; </h4>
        <div className="tooltip">
          <h3 onClick={handlerToggleBoard}>{currentBoard && currentBoard.title}</h3>
          <span className="tooltiptext">Nome do projeto</span>
        </div>
      </St.ProjectInfo>

      <St.UserInfo>
        <St.User>
          <h2>{user.name}</h2>
          <p>Online</p>
        </St.User>

        <St.ProfileIcon onClick={handleToggleMenu}>
          <img src={user.image ? user.image : icons.user1} alt="User icon" />
        </St.ProfileIcon>

        {toggleMenu && (
          <St.UserMenu>
            <St.MainUserMenu>
              <h2>Conta</h2>
              <img onClick={handleToggleMenu} src={icons.closeIcon} alt="close icon" />
            </St.MainUserMenu>

            <St.UserInfoMenu>
              <h2>{user.name}</h2>
              <h3>{user.email}</h3>
              <p>{user.about}</p>
            </St.UserInfoMenu>

            <St.MenuOption onClick={handleOpenEditProfile}>
              <p>Editar perfil</p>
            </St.MenuOption>

            <St.MenuOption onClick={handleLogout}>
              <p>Fazer logout</p>
            </St.MenuOption>

            <St.MenuOption onClick={handlerSaveBoard}>
              <p>Salvar board</p>
            </St.MenuOption>

            <St.MenuOption onClick={handleOpenEditBoard}>
              <p>Selecionar board</p>
            </St.MenuOption>
          </St.UserMenu>
        )}
      </St.UserInfo>
    </St.Bar>
  </St.TopContainer>
);

export default FixedMenu;
