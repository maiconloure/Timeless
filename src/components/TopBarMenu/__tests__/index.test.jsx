import TopBarMenu from '../index';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Snapshot', () => {
  it('renders unclicked', () => {
    const tree = renderer
      .create(
        <TopBarMenu
          values={{
            user: {
              name: 'Test',
              about: 'Testing test.',
              email: 'test@test.com',
              id: '0',
              image: 'test',
            },
            currentBoard: {
              data: 'test',
              description: 'Testing test.',
              id: '0',
              title: 'The Test',
              userId: '0',
              users: [{ name: 'Test', id: '0' }],
            },
            toggleMenu: false,
          }}
          handlers={{
            handleLogout: jest.fn(),
            handleOpenEditBoard: jest.fn(),
            handleOpenEditProfile: jest.fn(),
            handleToggleMenu: jest.fn(),
            handlerSaveBoard: jest.fn(),
            handlerToggleBoard: jest.fn(),
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
