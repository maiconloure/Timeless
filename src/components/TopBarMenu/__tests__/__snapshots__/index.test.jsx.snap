// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders unclicked 1`] = `
<div
  className="sc-bdnylx cCXgJk"
>
  <div
    className="sc-gtssRu cga-DwH"
  >
    <div
      className="sc-dlnjPT gtcHOL"
    >
      <a
        href="https://gitlab.com/capstoneproject-group-4/timeless"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          className="tooltip"
        >
          <img
            alt="Logo"
          />
          <span
            className="tooltiptext"
          >
            Timeless salva automaticamente o seu trabalho!
          </span>
        </div>
      </a>
      <div
        className="tooltip"
      >
        <h2>
           Kenzie Academy Brasil 
        </h2>
        <span
          className="tooltiptext"
        >
          Nome da empresa
        </span>
      </div>
      <h4>
           |   
      </h4>
      <div
        className="tooltip"
      >
        <h3
          onClick={[MockFunction]}
        >
          The Test
        </h3>
        <span
          className="tooltiptext"
        >
          Nome do projeto
        </span>
      </div>
    </div>
    <div
      className="sc-gKAblj kbRHGY"
    >
      <div
        className="sc-hKFyIo hNixbw"
      >
        <h2>
          Test
        </h2>
        <p>
          Online
        </p>
      </div>
      <div
        className="sc-eCApGN jbpBzz"
        onClick={[MockFunction]}
      >
        <img
          alt="User icon"
          src="test"
        />
      </div>
    </div>
  </div>
</div>
`;
